cmake/BLTPrivateMacros.cmake

```diff
@@ -272,6 +272,44 @@ macro(blt_inherit_target_info)
 
 endmacro(blt_inherit_target_info)
 
+##------------------------------------------------------------------------------
+## blt_expand_depends( DEPENDS_ON [dep1 ...]
+##                     RESULT [variable] )
+##------------------------------------------------------------------------------
+macro(blt_expand_depends)
+    set(options)
+    set(singleValueArgs RESULT)
+    set(multiValueArgs DEPENDS_ON)
+
+    # Parse the arguments
+    cmake_parse_arguments(arg "${options}" "${singleValueArgs}"
+                        "${multiValueArgs}" ${ARGN} )
+
+    # Expand dependency list
+    set(_already_proccessed_depends)
+    set(_expanded_DEPENDS_ON ${arg_DEPENDS_ON})
+    foreach( i RANGE 50 )
```

Comment:
Is "50" hard-coded here as a conservative upper limit, which will most likely never be exceeded?

Reply:
I believe so - this number was propagated from the existing logic in `blt_setup_target` added in #23, though the logic I added in this PR to ignore already-processed dependencies may allow this loop to be refactored and the hardcoded limit to be removed.